src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -195,18 +197,60 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
                 string telemetryPath = $"{featureFlagPath}:{FeatureManagementConstants.Telemetry}";
 
-                if (telemetry.Enabled)
-                {
-                    keyValues.Add(new KeyValuePair<string, string>($"{telemetryPath}:{FeatureManagementConstants.Enabled}", telemetry.Enabled.ToString()));
-                }
-
                 if (telemetry.Metadata != null)
                 {
                     foreach (KeyValuePair<string, string> kvp in telemetry.Metadata)
                     {
                         keyValues.Add(new KeyValuePair<string, string>($"{telemetryPath}:{FeatureManagementConstants.Metadata}:{kvp.Key}", kvp.Value));
                     }
                 }
+
+                if (telemetry.Enabled)
+                {
+                    byte[] featureFlagIdHash;
+
+                    using (HashAlgorithm hashAlgorithm = SHA256.Create())
+                    {
+                        featureFlagIdHash = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes($"{setting.Key}\n{(string.IsNullOrWhiteSpace(setting.Label) ? null : setting.Label)}"));
+                    }
+
+                    string featureFlagIdBase64 = Convert.ToBase64String(featureFlagIdHash);
+
+                    int indexOfEquals = featureFlagIdBase64.IndexOf("=");
+
+                    int stringBuilderCapacity = indexOfEquals != -1 ? indexOfEquals : featureFlagIdBase64.Length;
+
+                    StringBuilder featureFlagIdBuilder = new StringBuilder(stringBuilderCapacity);
+
+                    for (int i = 0; i < stringBuilderCapacity; i++)
+                    {
+                        if (featureFlagIdBase64[i] == '+')
+                        {
+                            featureFlagIdBuilder.Append('-');
+                        }
+                        else if (featureFlagIdBase64[i] == '/')
+                        {
+                            featureFlagIdBuilder.Append('_');
+                        }
+                        else
+                        {
+                            featureFlagIdBuilder.Append(featureFlagIdBase64[i]);
+                        }
+                    }
+
+                    keyValues.Add(new KeyValuePair<string, string>($"{telemetryPath}:{FeatureManagementConstants.Metadata}:{FeatureManagementConstants.FeatureFlagId}", featureFlagIdBuilder.ToString()));
+
+                    if (endpoint != null)
+                    {
+                        string featureFlagReference = $"{endpoint.AbsoluteUri}kv/{setting.Key}{(setting.Label != null ? $"?label={setting.Label}" : "")}";
```

Comment:
NIT: instead of null check, should we do string.IsNullOrWhiteSpace(setting.Label) to be consistent with what you have above?